[Cheng Chuan] iP

zhangchengchuan opened the pull request:
### KopitiamUncle

> "Teh O or Teh C” – A kopitiam uncle, probably

KopitiamUncle keeps tracks of your drinks **AND** your tasks.. It's,

* Uniquely Singaporean
* Chides you once in a while
* ~Easy~ **Extremely** easy to use and talk to

All you need to do is,

1. Download it from [here](https://github.com/zhangchengchuan/ip/releases/tag/A-Jar).
1. Run the programme
1. Talk to the uncle.
1. Be amazed! 🤯 

And it is **FREE**!

Features:

- [X]  Managing tasks
- [X]  Managing deadlines
- [ ]  Reminders (coming soon)

If you Java programmer, you can use it to practice Java too. Here's the main method:

```java
public class Main {
    public static void main(String[] args) {
        new Duke("./tasks.txt").run();
    }
}
```

Commit:
Add Gradle support

zhangchengchuan committed:
Level-1

zhangchengchuan committed:
modified:   main/java/Duke.java

zhangchengchuan committed:
modified:   main/java/Duke.java
	modified:   main/java/TodoList.java

zhangchengchuan committed:
modified:   main/java/Duke.java
	modified:   main/java/TodoList.java

zhangchengchuan committed:
modified:   main/java/Duke.java
	new file:   main/java/Task.java
	modified:   main/java/TodoList.java

zhangchengchuan committed:
new file:   main/java/Deadline.java
	modified:   main/java/Duke.java
	new file:   main/java/Event.java
	modified:   main/java/Task.java
	new file:   main/java/Todo.java
	modified:   main/java/TodoList.java

zhangchengchuan committed:
modified:   EXPECTED.TXT
	modified:   input.txt
	modified:   ../src/main/java/Deadline.java
	modified:   ../src/main/java/Duke.java
	modified:   ../src/main/java/Event.java
	modified:   ../src/main/java/Task.java
	modified:   ../src/main/java/Todo.java
	modified:   ../src/main/java/TodoList.java

zhangchengchuan committed:
modified:   src/main/java/Deadline.java
	modified:   src/main/java/Duke.java
	modified:   src/main/java/Event.java
	modified:   src/main/java/Task.java
	modified:   src/main/java/Todo.java
	modified:   src/main/java/TodoList.java

zhangchengchuan committed:
modified:   src/main/java/Duke.java
	new file:   src/main/java/DukeExceptions.java
	new file:   src/main/java/IndexNotInListException.java
	new file:   src/main/java/NoDescriptionException.java
	new file:   src/main/java/NonsenseException.java
	modified:   src/main/java/TodoList.java

zhangchengchuan committed:
modified:   src/main/java/Duke.java
	modified:   src/main/java/TodoList.java

zhangchengchuan committed:
new file:   src/main/java/DukeException.java
	deleted:    src/main/java/DukeExceptions.java
	modified:   src/main/java/IndexNotInListException.java
	modified:   src/main/java/NoDescriptionException.java
	modified:   src/main/java/NonsenseException.java

zhangchengchuan committed:
modified:   src/main/java/Duke.java
	modified:   src/main/java/TodoList.java

zhangchengchuan committed:
new file:   src/main/java/DateManagement.java
	modified:   src/main/java/Deadline.java
	modified:   src/main/java/Duke.java
	modified:   src/main/java/Event.java
	modified:   src/main/java/Task.java
	modified:   src/main/java/TodoList.java
	new file:   src/main/java/WrongInputException.java

zhangchengchuan committed:
Merge branch 'branch-level-8'

	new file:   src/main/java/DateManagement.java
	modified:   src/main/java/Deadline.java
	modified:   src/main/java/Duke.java
	modified:   src/main/java/Event.java
	modified:   src/main/java/Task.java
	modified:   src/main/java/TodoList.java
	new file:   src/main/java/WrongInputException.java

zhangchengchuan committed:
new file:   src/main/java/Command.java
	deleted:    src/main/java/DateManagement.java
	modified:   src/main/java/Deadline.java
	modified:   src/main/java/Duke.java
	modified:   src/main/java/Event.java
	renamed:    src/main/java/DukeException.java -> src/main/java/Exceptions/DukeException.java
	renamed:    src/main/java/IndexNotInListException.java -> src/main/java/Exceptions/IndexNotInListException.java
	renamed:    src/main/java/NoDescriptionException.java -> src/main/java/Exceptions/NoDescriptionException.java
	new file:   src/main/java/Exceptions/NoPreviousFileException.java
	renamed:    src/main/java/NonsenseException.java -> src/main/java/Exceptions/NonsenseException.java
	renamed:    src/main/java/WrongInputException.java -> src/main/java/Exceptions/WrongInputException.java
	new file:   src/main/java/Parser.java
	new file:   src/main/java/Storage.java
	modified:   src/main/java/Task.java
	new file:   src/main/java/TaskList.java
	modified:   src/main/java/Todo.java
	deleted:    src/main/java/TodoList.java
	new file:   src/main/java/Ui.java

zhangchengchuan committed:
renamed:    src/main/java/Command.java -> src/main/java/Duke/Command/Command.java
	renamed:    src/main/java/Duke.java -> src/main/java/Duke/Duke.java
	renamed:    src/main/java/Exceptions/DukeException.java -> src/main/java/Duke/Exceptions/DukeException.java
	renamed:    src/main/java/Exceptions/IndexNotInListException.java -> src/main/java/Duke/Exceptions/IndexNotInListException.java
	renamed:    src/main/java/Exceptions/NoDescriptionException.java -> src/main/java/Duke/Exceptions/NoDescriptionException.java
	renamed:    src/main/java/Exceptions/NoPreviousFileException.java -> src/main/java/Duke/Exceptions/NoPreviousFileException.java
	renamed:    src/main/java/Exceptions/NonsenseException.java -> src/main/java/Duke/Exceptions/NonsenseException.java
	renamed:    src/main/java/Exceptions/WrongInputException.java -> src/main/java/Duke/Exceptions/WrongInputException.java
	renamed:    src/main/java/Parser.java -> src/main/java/Duke/Parser/Parser.java
	renamed:    src/main/java/Storage.java -> src/main/java/Duke/Storage/Storage.java
	renamed:    src/main/java/Deadline.java -> src/main/java/Duke/Tasks/Deadline.java
	renamed:    src/main/java/Event.java -> src/main/java/Duke/Tasks/Event.java
	renamed:    src/main/java/Task.java -> src/main/java/Duke/Tasks/Task.java
	renamed:    src/main/java/TaskList.java -> src/main/java/Duke/Tasks/TaskList.java
	renamed:    src/main/java/Todo.java -> src/main/java/Duke/Tasks/Todo.java
	renamed:    src/main/java/Ui.java -> src/main/java/Duke/Ui/Ui.java
	modified:   tasks.txt

zhangchengchuan committed:
new file:   src/test/java/Duke/DeadlineTest.java
	new file:   src/test/java/Duke/DukeTest.java
	new file:   src/test/java/Duke/EventTest.java
	modified:   tasks.txt

zhangchengchuan committed:
new file:   src/main/java/META-INF/MANIFEST.MF

zhangchengchuan committed:
modified:   src/main/java/Duke/Command/Command.java
	modified:   src/main/java/Duke/Duke.java
	modified:   src/main/java/Duke/Parser/Parser.java
	modified:   src/main/java/Duke/Storage/Storage.java
	modified:   src/main/java/Duke/Tasks/TaskList.java
	modified:   tasks.txt

zhangchengchuan committed:
modified:   src/main/java/Duke/Duke.java
	modified:   src/main/java/Duke/Parser/Parser.java
	modified:   src/main/java/Duke/Storage/Storage.java
	modified:   src/main/java/Duke/Tasks/Deadline.java
	modified:   src/main/java/Duke/Tasks/Event.java
	modified:   src/main/java/Duke/Tasks/TaskList.java

zhangchengchuan committed:
modified:   src/main/java/Duke/Command/Command.java
	modified:   src/main/java/Duke/Parser/Parser.java
	modified:   src/main/java/Duke/Tasks/Task.java
	modified:   src/main/java/Duke/Tasks/TaskList.java
	modified:   src/main/java/Duke/Ui/Ui.java

zhangchengchuan committed:
Merge branch 'branch-A-JavaDoc'

	modified:   src/main/java/Duke/Command/Command.java
	modified:   src/main/java/Duke/Duke.java
	modified:   src/main/java/Duke/Parser/Parser.java
	modified:   src/main/java/Duke/Storage/Storage.java
	modified:   src/main/java/Duke/Tasks/TaskList.java

zhangchengchuan committed:
Merge branch 'branch-A-CodingStandard'

	modified:   src/main/java/Duke/Duke.java
	modified:   src/main/java/Duke/Parser/Parser.java
	modified:   src/main/java/Duke/Storage/Storage.java
	modified:   src/main/java/Duke/Tasks/Deadline.java
	modified:   src/main/java/Duke/Tasks/Event.java
	modified:   src/main/java/Duke/Tasks/TaskList.java

zhangchengchuan committed:
Merge remote-tracking branch 'origin/add-gradle-support' into branch-A-Gradle

zhangchengchuan committed:
modified:   README.md
	modified:   build.gradle
	new file:   config/checkstyle/checkstyle.xml
	new file:   config/checkstyle/suppressions.xml
	modified:   src/main/java/META-INF/MANIFEST.MF
	renamed:    src/main/java/Duke/Command/Command.java -> src/main/java/duke/Command/Command.java
	renamed:    src/main/java/Duke/Duke.java -> src/main/java/duke/Duke.java
	renamed:    src/main/java/Duke/Exceptions/DukeException.java -> src/main/java/duke/Exceptions/DukeException.java
	renamed:    src/main/java/Duke/Exceptions/IndexNotInListException.java -> src/main/java/duke/Exceptions/IndexNotInListException.java
	renamed:    src/main/java/Duke/Exceptions/NoDescriptionException.java -> src/main/java/duke/Exceptions/NoDescriptionException.java
	renamed:    src/main/java/Duke/Exceptions/NoPreviousFileException.java -> src/main/java/duke/Exceptions/NoPreviousFileException.java
	renamed:    src/main/java/Duke/Exceptions/NonsenseException.java -> src/main/java/duke/Exceptions/NonsenseException.java
	renamed:    src/main/java/Duke/Exceptions/WrongInputException.java -> src/main/java/duke/Exceptions/WrongInputException.java
	renamed:    src/main/java/Duke/Parser/Parser.java -> src/main/java/duke/Parser/Parser.java
	renamed:    src/main/java/Duke/Storage/Storage.java -> src/main/java/duke/Storage/Storage.java
	renamed:    src/main/java/Duke/Tasks/Deadline.java -> src/main/java/duke/Tasks/Deadline.java
	renamed:    src/main/java/Duke/Tasks/Event.java -> src/main/java/duke/Tasks/Event.java
	renamed:    src/main/java/Duke/Tasks/Task.java -> src/main/java/duke/Tasks/Task.java
	renamed:    src/main/java/Duke/Tasks/TaskList.java -> src/main/java/duke/Tasks/TaskList.java
	renamed:    src/main/java/Duke/Tasks/Todo.java -> src/main/java/duke/Tasks/Todo.java
	renamed:    src/main/java/Duke/Ui/Ui.java -> src/main/java/duke/Ui/Ui.java
	renamed:    src/test/java/Duke/DeadlineTest.java -> src/test/java/duke/DeadlineTest.java
	renamed:    src/test/java/Duke/DukeTest.java -> src/test/java/duke/DukeTest.java
	renamed:    src/test/java/Duke/EventTest.java -> src/test/java/duke/EventTest.java
	modified:   text-ui-test/runtest.bat

darren2pro commented on src/main/java/Duke/Command/Command.java:
```diff
@@ -0,0 +1,294 @@
+    }
+
+    public static class DoneCommand extends Command {
+        String input;
```
Perhaps you would like to define an access modifier here? But it's just for clarity, this should default to package-private.

darren2pro commented on src/main/java/Duke/Parser/Parser.java:
```diff
@@ -0,0 +1,39 @@
+     */
+    public static Command parse(String stringCommand) {
+        String keyword = stringCommand.split(" ")[0].trim().toLowerCase();
+        switch (keyword) {
```
Just a gentle reminder, do remember to put "switch" and "case" on the same indentation level. 👍

darren2pro reviewed (commented):
LGTM. Just some nits to fix.

zhangchengchuan committed:
modified:   build.gradle
	modified:   src/main/java/duke/Command/Command.java
	modified:   src/main/java/duke/Parser/Parser.java
	modified:   src/main/java/duke/Storage/Storage.java
	modified:   src/main/java/duke/Tasks/Event.java
	modified:   src/test/java/duke/DeadlineTest.java

zhangchengchuan committed:
modified:   build.gradle
	modified:   src/main/java/duke/Duke.java
	new file:   src/main/java/duke/Gui/DialogBox.java
	new file:   src/main/java/duke/Gui/Main.java
	new file:   src/main/java/duke/Gui/MainWindow.java
	new file:   src/main/java/duke/Launcher.java
	new file:   src/main/resources/images/guy.jpg
	new file:   src/main/resources/images/kopitiam uncle.jpg
	new file:   src/main/resources/view/DialogBox.fxml
	new file:   src/main/resources/view/MainWindow.fxml
	modified:   tasks.txt

zhangchengchuan committed:
Update README.md

RachelCheah commented on src/main/java/duke/Command/Command.java:
```diff
@@ -0,0 +1,296 @@
+
+    }
+
+    public static class ClearCommand extends Command {
```
I think you should split this class into multiple files as each top-level class should reside in a source file of its own!

RachelCheah reviewed (commented):
LGTM! The code seems to adhere to the coding standard specified overall!

RachelCheah commented on src/main/java/duke/Command/Command.java:
```diff
@@ -0,0 +1,296 @@
+package duke.Command;
+
+import java.util.ArrayList;
```
I would suggest listing the imports from java first to follow checkstyle!

dannylty commented on src/main/java/duke/Command/Command.java:
```diff
@@ -0,0 +1,296 @@
+
+import java.util.ArrayList;
+
+import duke.Exceptions.DukeException;
```
As per the coding standards, package names should be all lower case.

dannylty commented on src/main/java/duke/Storage/Storage.java:
```diff
@@ -0,0 +1,127 @@
+        BufferedWriter bw;
+        try {
+            bw = new BufferedWriter(new FileWriter(path));
+            for (Task t : tasklist.getTaskList()) {
```
I noticed that in TaskList.java line 74, you used `Task t: taskList`. I think standardizing it to have whitespace around the colon is good.

dannylty reviewed (commented):
I like how you put the String message literals into one class.

zhangchengchuan committed:
Add assertions into multiple classes

Assertions have been added to multiple parts of the code base.

Most of these assertions are to check if the respective commands or strings are empty, which they should not be.

The commands or strings checked should not be empty because:
* Duke Exception should have caught those errors
* If the user inputs are empty, the programme will not produce any outputs or call any functions

zhangchengchuan committed:
Merge pull request #1 from zhangchengchuan/branch-A-Assertions

Add assertions into multiple classes

zhangchengchuan committed:
Code change to follow required Code Quality

Changes were made to several files, functions and variable names to be in line with required Code Quality.

zhangchengchuan committed:
Merge pull request #2 from zhangchengchuan/branch-A-CodeQuality

Code change to follow required Code Quality

zhangchengchuan committed:
Add Extension B-Snooze

Snooze function has been added. Just type <Snooze [index of task]> to postpone it by 1 week.

zhangchengchuan committed:
Set theme jekyll-theme-cayman

zhangchengchuan committed:
Add User Guide and fixed some buggy code

Simple User guide for Kopitiam Uncle Chatbot has been added to teach new users how to use it. Previously buggy event/deadline tasks have been fixed as well.

zhangchengchuan committed:
Merge branch 'master' of https://github.com/zhangchengchuan/ip

zhangchengchuan committed:
Fix bug with regards to javafx
